model/UserModel.ts

```diff
@@ -1,5 +1,153 @@
+import type { UidIdentifier } from "firebase-admin/auth";
+
 export class UserModel {
-    /**
-     * Model for the user
-     */
-}
\ No newline at end of file
+    //private fields for the user's statistics
+
+    //name of the user
+    private _userName: string = '';
```

Comment:
```suggestion
    private _username: string = '';
```
No need to capitalize `Name` here.